packages/test/test-service-load/src/nodeStressTest.ts

```diff
@@ -138,13 +166,93 @@ async function orchestratorProcess(
                     env: envVar,
                 },
             );
+
+            if (args.enableMetrics === true) {
+                setupTelemetry(runnerProcess, logger, index, username);
+            }
+
             return new Promise((resolve) => runnerProcess.once("close", resolve));
         }));
-    } finally{
+    } finally {
+        if (logger !== undefined) {
+            await logger.flush();
+        }
         await safeExit(0, url);
     }
 }
 
+/**
+ * Setup event and metrics telemetry to be sent to loggers.
+ */
+function setupTelemetry(
+    process: child_process.ChildProcess,
+    logger: ITelemetryLogger,
+    runId: number,
+    username?: string) {
+    logger.send({
+        category: "metric",
+        eventName: "Runner Started",
+        value: 1,
+        username,
+        runId,
+    });
+
+    process.once("error", (e) => {
+        logger.send({
+            category: "metric",
+            eventName: "Runner Start Error",
+            value: 1,
```

Comment:
I feel like errors should not use "metric" category. They should use "error" category, but they can use some other property to indicate some special treatment required. I.e.
   errorDetectedByTestHarness: true,

Reply:
Note - independent of that, I'd recommend to use sendTelemetryEvent({ ... }, error) that would properly format error and extract all the interesting fields from error object no matter if it's of type Error or not.

Reply:
This is to capture the metrics of how many error occurred to be drawn on metrics graphs. We send these to app insights with  trackMetrics() compared to other event logging which is trackEvent() which will require custom queries to be plotted for dashboard.

Reply:
Separated metrics and logging of error details.
Logging error details as events using sendTelemetryEvent.